src/main/java/io/omnition/loadgenerator/util/TraceGenerator.java

```diff
@@ -36,36 +38,62 @@ private TraceGenerator(Topology topology) {
-                .map(t -> KeyValue.ofStringType(t.getKey(), t.getValue())).collect(Collectors.toList());
-        span.tags.addAll(tags);
+
+        // Setup base tags
+        span.setHttpMethodTag("GET");
+        span.setHttpUrlTag("http://" + serviceTier.serviceName + routeName);
+        // Get additional tags for this route, and update with any inherited tags
+        TagSet routeTags = serviceTier.getTagSet(routeName);
+        HashMap<String, Object> tagsToSet = new HashMap<>(routeTags.tags);
+        if (parentTagSet != null && routeTags.inherit != null) {
+            for (String inheritTagKey : routeTags.inherit) {
+                Object value = parentTagSet.tags.get(inheritTagKey);
+                if (value != null) {
+                    tagsToSet.put(inheritTagKey, value);
+                }
+            }
+        }
+
+        // Set the additional tags on the span
+        List<KeyValue> spanTags = tagsToSet.entrySet().stream()
+            .map(t -> {
+                Object val = t.getValue();
+                if (val instanceof String) {
+                    return KeyValue.ofStringType(t.getKey(), (String) val);
+                }
+                if (val instanceof Double) {
+                    return KeyValue.ofLongType(t.getKey(), ((Double) val).longValue());
+                }
+                if (val instanceof Boolean) {
+                    return KeyValue.ofBooleanType(t.getKey(), (Boolean) val);
+                }
+                return null;
+            })
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+        span.tags.addAll(spanTags);
 
         final AtomicLong maxEndTime = new AtomicLong(startTimeMicros);
-        if (random.nextInt(100) < om.errorPercent) {
-            // inject error and terminate trace there
-            span.markError();
+        if (tagsToSet.get(SpanConventions.IS_ERROR_KEY) != null) {
+            // inject root cause error and terminate trace there
             span.markRootCauseError();
-            span.setHttpCode(om.errorCode);
         } else {
             // no error, make downstream calls
             route.downstreamCalls.forEach((s, r) -> {
-                long childStartTimeMicros = startTimeMicros + TimeUnit.MILLISECONDS.toMicros(random.nextInt(om.maxLatencyMillis));
+                long childStartTimeMicros = startTimeMicros + TimeUnit.MILLISECONDS.toMicros(random.nextInt(route.maxLatencyMillis));
                 ServiceTier childSvc = this.topology.getServiceTier(s);
-                Span childSpan = createSpanForServiceRouteCall(childSvc, r, childStartTimeMicros);
+                Span childSpan = createSpanForServiceRouteCall(routeTags, childSvc, r, childStartTimeMicros);
                 Reference ref = new Reference(RefType.CHILD_OF, span.id, childSpan.id);
                 childSpan.refs.add(ref);
                 maxEndTime.set(Math.max(maxEndTime.get(), childSpan.endTimeMicros));
```

Comment:
ok github is lame but the lines below may not always work as expected anymore - perhaps we need to generalize the fragment below as childSpan.isErrorSpan()
and make an isErrorSpan method on Span class that checks both isError tag as well as HTTP error code. If either condition matches then we return true.